model/roots.js

```diff
@@ -51,9 +51,31 @@ async function getVerseRootWords(data) {
         })
     );
 }
+async function getRootsMeaning(data){
+    const sentences = [];
+    var all_roots = getVerseRootWords(data);
+    for (let root of all_roots){
```

Comment:
So you've called `getVerseRootWords` correctly, but the for loop should be going over the data array (i.e. `all_roots.data`). You should also be checking to ensure that the response from `getVerseRootWords` is successful (i.e. `all_roots.success`)

Reply:
Also, you forgot to add an `await` on the `getVerseRootWords` call since it's an asynchronous function.